request.js

```diff
@@ -1364,6 +1364,8 @@ Request.prototype.aws = function (opts, now) {
       host: self.uri.host,
       path: self.uri.path,
       method: self.method,
+      service: opts.service,
```

Comment:
I'm not positive what `service` and `region` might be used for other than API Gateway. API Gateway only supports version 4 though.